uReplicator-Controller/src/main/java/com/uber/stream/kafka/mirrormaker/controller/core/ManagerControllerHelix.java

```diff
@@ -68,6 +71,16 @@ public synchronized void start() {
         new ControllerStateModelFactory(this));
     try {
       _helixZkManager.connect();
+      InstanceConfig instanceConfig = new InstanceConfig(_instanceId);
+      instanceConfig.setHostName(_hostname);
+      ZkHelixPropertyStore<ZNRecord> propertyStore = _helixZkManager.getHelixPropertyStore();
+      String resourcePath = Constants.CONTROLLER_ID_HOSTNAME_PROPERTY_KEY;
+      ZNRecord znRecord = propertyStore.get(resourcePath, null, AccessOption.PERSISTENT);
+      if (znRecord == null) {
+        znRecord = new ZNRecord(resourcePath);
+      }
+      znRecord.setSimpleField(_instanceId, _hostname);
+      propertyStore.set(resourcePath, znRecord, AccessOption.PERSISTENT);
```

Comment:
Looks like it's setting the whole znRecord in each controller instead of incremental update, there might be race condition between different controllers.

Reply:
Also is there any place which might use the id->hostname after controller is down? Currently we don't remove the mapping after controller is down.

Reply:
good point